documentation/modules/con-common-configuration-properties.adoc

```diff
@@ -310,37 +310,96 @@ You can specify which container image to use for a component using the `image` p
+
+.Values specified for image and Kafka version
+[cols="1,1,2",options="header"]
+|===
+| `image` set?    | `version` set?  | Result
+
+| ✗           | ✗         | Uses Cluster Operator's default image and corresponding Kafka version
+| ✓           | ✗         | Uses specified image and default Kafka version
+| ✗           | ✓         | Uses image from environment variable for specified Kafka version
+| ✓           | ✓         | Uses specified image and assumes specified Kafka version matches
+|===
+
+NOTE: To avoid Kafka version and image mismatches, set the `version` property and allow the Cluster Operator to select the matching image from its mappings. 
+If you need to change the default image mapping for a given Kafka version, configure the Cluster Operator’s environment variables.
+
+Even if the configuration is syntactically correct, it can still be invalid if the image and version mismatch.
+To ensure a valid configuration:
+
+* The specified `version` *must* match the Kafka version that the image is built for.
+* The specified `version` must be one of the versions supported by the operator.
+* If you set a custom `image`, always set `version` to the Kafka version of that image.
+
+Here we can see what happens with versions {KafkaVersionLower} and {DefaultKafkaVersion} (the default).
+
+.Valid and invalid configuration examples
+[cols="1,1,1,1,1",options="header"]
+|===
+| `image` set? | `version` set? | Image used | Kafka version | Valid?
+
+| ✗ | ✗ | {DefaultKafkaVersion} (default) | {DefaultKafkaVersion} | ✓
+| ✗ | {KafkaVersionLower} | {KafkaVersionLower} (from mapping) | {KafkaVersionLower} | ✓
+| Custom {KafkaVersionLower} | {KafkaVersionLower} | Custom {KafkaVersionLower} | {KafkaVersionLower} | ✓
+| Custom {KafkaVersionLower} | ✗ | Custom {KafkaVersionLower} | {DefaultKafkaVersion} (default) | ✗
+| Custom {KafkaVersionLower} | {DefaultKafkaVersion} | Custom {KafkaVersionLower} | {DefaultKafkaVersion} | ✗
+|===
 
-* If `image` is given but `version` is not, then the given image is used and the `version` is assumed to be the Cluster Operator's default Kafka version.
+NOTE: _Custom {KafkaVersionLower}_ refers to a custom user-provided container image built against Kafka {KafkaVersionLower}.
+Setting a *custom image* means you set the `image` property to a user-provided image. 
+The operator does not automatically change this value during upgrades.
 
-* If `version` is given but `image` is not, then the image that corresponds to the given version in the environment variable is used.
+*Handling upgrades with custom images*
 
-* If both `version` and `image` are given, then the given image is used. The image is assumed to contain a Kafka image with the given version.
+When using custom images, upgrades must keep the `image` and `version` in sync.
+The Cluster Operator cannot update custom images automatically during an upgrade.
```

Comment:
I think here we need to explain a bit more that this applies only when the custom image is set through the `image` field in the CR. It does not apply when:
* KafkaConnect Build is used
* The image from the `STRIMZI_KAFKA_CONNECT_IMAGES` variable is used